File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/tv/series/model/TvSeriesAggregateCastRole.java
package dev.reuss.tmdb.domain.tv.series.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import dev.reuss.tmdb.common.TmdbModel;

/**
 * Aggregate cast role for a TMDB TV series.
 *
 * @param creditId     TMDB credit id
 * @param character    character name
 * @param episodeCount episode count
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record TvSeriesAggregateCastRole(
        @JsonProperty("credit_id")
        String creditId,

        String character,

        @JsonProperty("episode_count")
        int episodeCount
) implements TmdbModel {
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/tv/series/model/TvSeriesAggregateCastRole.kt
package dev.reuss.tmdb.domain.tv.series.model

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel

/**
 * Aggregate cast role for a TMDB TV series.
 *
 * @property creditId TMDB credit id
 * @property character character name
 * @property episodeCount episode count
 */
@JvmRecord
data class TvSeriesAggregateCastRole(
    @all:JsonProperty("credit_id")
    val creditId: String?,

    @all:JsonProperty("character")
    val character: String?,

    @all:JsonProperty("episode_count")
    val episodeCount: Int
) : TmdbModel